### ISSUE TITLE
Counting Cards is unclear or too difficult

### ISSUE THREAD

**Issue by QuincyLarson** [freeCodeCamp/freeCodeCamp#15615]
According to Google Analytics, the [Counting Cards challenge](https://www.freecodecamp.com/challenges/counting-cards) is way too hard, and is taking campers an average of 8 minutes to finish (if they finish - a lot of them seem to give up). 

According to this table, campers spend more time on it than they do on the algorithm challenges.

![](https://www.evernote.com/l/Ar9YCZe5amdJxo7PxclM03fMCHcDIQYSKZwB/image.png)

We need to either reword this or simplify the challenge so it can be completed in closer to 3 minutes. Otherwise, we should move it back to the algorithm section.

**Comment by Margaret2** [comment]
This is a hard one, I think one of the main issues people have is that they don't understand this challenge needs a switch statement **and** an if statement. The instructions just say, "The function will then return a string..." It's not clear that there are 2 steps required, so people are always putting the return statements inside the switch statement.

**Comment by QuincyLarson** [comment]
@Margaret2 Thanks for your feedback. Do you think we need to move it further back in the section?  Or perhaps to the algorithm section?

**Comment by prateekgoel** [comment]
@QuincyLarson I think we should move it after the "Profile Lookup" challenge in the same section.

**Comment by QuincyLarson** [comment]
@prateekgoel we are discussing moving Profile Lookup to the algorithms section: https://github.com/freeCodeCamp/freeCodeCamp/issues/15616

Do you think we should move this challenge there as well?

**Comment by prateekgoel** [comment]
@QuincyLarson I don't think we should move Profile Lookup to algorithms section because it's more of a basic Javascript to learn and not any sort of algorithm.
Moreover, as discussed in https://github.com/freeCodeCamp/freeCodeCamp/issues/15616 , rewording the instructions would make Profile Lookup challenge simpler and easy to understand.

**Comment by QuincyLarson** [comment]
@pra85 OK - I noticed you claimed #15616. That sounds good. So once you fix the wording on that, perhaps you can also see if there's a way to make Counting Cards easier or more clear. Then we can move Counting Cards to immediately follow Profile Lookup.

**Comment by prateekgoel** [comment]
@QuincyLarson I am sorry but was the above comment for me?

**Comment by QuincyLarson** [comment]
@prateekgoel Yes - it's for you. I autocompleted the wrong username :)

**Comment by prateekgoel** [comment]
Ok Thanks. I'll do that.

**Comment by prateekgoel** [comment]
@QuincyLarson - I think @Marmiz is working on https://github.com/freeCodeCamp/freeCodeCamp/issues/15616  as per the discussion.
So once that is done, i'll have a look at Counting Cards challenge

**Comment by QuincyLarson** [comment]
@prateekgoel OK - sounds good :)

**Comment by prateekgoel** [comment]
@QuincyLarson This issue targets the current FCC and not the beta one, right?
Because I see beta already have an option to create topic on forum under the Bug Button functionality.
And the layout described in issue belongs to current FCC.
So I the code resides in backup/master branch, right?
Also, can you please help me location the files i need to work on?

**Comment by Manish-Giri** [comment]
@prateekgoel Counting cards for `backup/master` is [here](https://github.com/freeCodeCamp/freeCodeCamp/blob/backup/master/seed/challenges/01-front-end-development-certification/basic-javascript.json#L3797).

**Comment by prateekgoel** [comment]
Thanks @Manish-Giri . But I mistakenly commented on wrong issue. That was for another issue.

**Comment by QuincyLarson** [comment]
@prateekgoel I think @Manish-Giri is no handling #15616 - could you go ahead and start working on this issue in parallel?

**Comment by prateekgoel** [comment]
@QuincyLarson Sure. I'll start working on this.

**Comment by prateekgoel** [comment]
@QuincyLarson I reworded the challenge instructions below the table to make it easy to understand. Please have a look and share your thoughts. Then, I'll submit the PR.

....

Write a card counting function which will receive a `card` parameter. Increment or decrement the value of the global `count` variable according to the value of card parameter. Refer the above table to see how much the `count` variable should be incremented/decremented for a particular value of card parameter.
The function should return a string with the current count value joined to another string `"Bet"` if the count is positive, or `"Hold"` if the count is zero or negative. Separate the current count and the player's decision (`"Bet"` or `"Hold"`) by a single space.

### Example Output
`"-3 Hold"`
`"5 Bet"`

### Here are some helpful links:
[Switch Statement](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch)

### Hint

....

**Comment by QuincyLarson** [comment]
@prateekgoel We shouldn't link to an external resources this early in the curriculum. We should instead explain what we need to explain ourselves.

After looking at this, I'm now convinced that this challenge is too hard. It's asking the camper to do too many things. We should move it to the basic algorithms section instead.

**Comment by prateekgoel** [comment]
@QuincyLarson I thought to link to external resource as we already have link to the same external resource few challenges before this one.

Still, if you want i can remove the link and include the explanation on using the `switch` statement.

Regarding moving this challenge to basic algorithms section, I'm still not sure if this would be a good idea as this challenge doesn't seem to match other challenges in the basic algorithms section. I mean it's more of basic JavaScript thing which uses simple `switch` statement. How about moving it down in the same section?

Otherwise, I'm fine with moving it to Basic Algorithms section. I am sure you've thought enough and came to this decision.

Let me know your views regarding this. Then, I'll work accordingly and submit the PR.

**Comment by QuincyLarson** [comment]
@prateekgoel If we could figure out a way to reduce the number of things campers need to do to pass this challenge, we could keep it where it is. We spend a lot of time talking about switch statements leading up to this challenge. Can you try and figure out a way to reduce the complexity of the challenge?

**Comment by prateekgoel** [comment]
@QuincyLarson Sure. I will try it and let you know.

**Comment by prateekgoel** [comment]
Hello @QuincyLarson ,
I've sent you a message on Gitter. Please have a look when you are free.

**Comment by QuincyLarson** [comment]
@prateekgoel I've responded to you on Gitter.

**Comment by QuincyLarson** [comment]
This challenge could still use a redesign to make it clearer and easier for a beginner to complete.

**Comment by Manish-Giri** [comment]
I'll take this up.

**Comment by iamlatin** [comment]
im watching

**Comment by iamlatin** [comment]
i cant do any code

**Comment by Nirajn2311** [comment]
@Manish-Giri Any progress in this?

**Comment by sprasih1** [comment]
@Manish-Giri  @QuincyLarson 
We can replace the following statement this sounded confusing to me :

"Having more high cards remaining in the deck favors the player. Each card is assigned a value according to the table below. When the count is positive, the player should bet high. When the count is zero or negative, the player should bet low."

with

"Each card is assigned a value according to the table below. When the relative count is positive, the player should bet. When the relative count is zero or negative, the player should hold".
OR
"Each card is assigned a value according to the table below. When the count is positive, the player should bet. When the count is zero or negative, the player should hold".

**Comment by Manish-Giri** [comment]
Just getting back on this. Will post an update soon.

**Comment by thecodingaviator** [comment]
@Manish-Giri Anything yet?

**Comment by QuincyLarson** [comment]
@thecodingaviator @Manish-Giri Since we're replacing the JavaScript curriculum soon, I'm going to go ahead and close this issue so.

@Manish-Giri We would welcome your experienced hands with our new project-oriented curriculum: https://www.freecodecamp.org/forum/t/help-us-build-version-7-0-of-the-freecodecamp-curriculum/263546

**Comment by Manish-Giri** [comment]
@QuincyLarson Thanks for the update! I'll be happy to help out with the new curriculum. I'm closing this issue for now.
